packages/site/src/components/Navigation/TableOfContentsItems.tsx

```diff
@@ -35,10 +35,16 @@ function nestToc(toc: Heading[]): NestedHeading[] {
   return items;
 }
 
+function pathnameMatchesHeading(pathname: string, heading: Heading, baseurl?: string) {
+  const headingPath = withBaseurl(heading.path, baseurl);
+  if (pathname && headingPath === `${pathname}/index`) return true;
```

Comment:
There's a few small changes in here that allow `index` pages to be accessible at the folder-path url.

For example, the file `folder1/folder2/index.md` is available at `/folder1/folder2` (and `/folder1/folder2/index` redirects back to `/folder1/folder2`).